l1-contracts/contracts/bridge/L1WethBridge.sol

```diff
@@ -151,42 +340,151 @@ contract L1WethBridge is IL1Bridge, ReentrancyGuard {
+
+            // If the refund recipient is not specified, the refund will be sent to the sender of the transaction.
+            // Otherwise, the refund will be sent to the specified address.
+            // If the recipient is a contract on L1, the address alias will be applied.
+            address refundRecipient = _refundRecipient;
+            if (_refundRecipient == address(0)) {
+                refundRecipient = msg.sender != tx.origin
+                    ? AddressAliasHelper.applyL1ToL2Alias(msg.sender)
+                    : msg.sender;
+            }
+            txHash = _depositSendTx(
+                _chainId,
+                _mintValue,
+                _amount,
+                l2TxCalldata,
+                _l2TxGasLimit,
+                _l2TxGasPerPubdataByte,
+                refundRecipient
+            );
+        }
+        emit DepositInitiatedSharedBridge(_chainId, txHash, msg.sender, _l2Receiver, _l1Token, _amount);
+    }
 
-        emit DepositInitiated(txHash, msg.sender, _l2Receiver, _l1Token, _amount);
+    function _depositSendTx(
+        uint256 _chainId,
+        uint256 _mintValue,
+        uint256 _amount,
+        bytes memory _l2TxCalldata,
+        uint256 _l2TxGasLimit,
+        uint256 _l2TxGasPerPubdataByte,
+        address _refundRecipient
+    ) internal returns (bytes32 txHash) {
+        // we don't save the depositAmount because base asset is sent to refundrecipient
+
+        L2TransactionRequestDirect memory request = L2TransactionRequestDirect({
+            chainId: _chainId,
+            l2Contract: l2BridgeAddress[_chainId],
+            mintValue: _mintValue,
+            l2Value: _amount,
+            l2Calldata: _l2TxCalldata,
+            l2GasLimit: _l2TxGasLimit,
+            l2GasPerPubdataByteLimit: _l2TxGasPerPubdataByte,
+            factoryDeps: new bytes[](0),
+            refundRecipient: _refundRecipient
+        });
+        txHash = bridgehub.requestL2Transaction{value: _mintValue}(request);
+    }
+
+    // we have to keep track of bridgehub deposits to track each chain's assets
+    function bridgehubDepositBaseToken(
+        uint256 _chainId,
+        address ,//_prevMsgSender,
+        address _token,
+        uint256 _amount
+    ) external payable override onlyBridgehubOrEthChain(_chainId) {
+        require(_token == ETH_TOKEN_ADDRESS, "L1WETHBridge: Invalid token");
+        require(msg.value == _amount, "L1WETHBridge: msg.value not equal to amount");
+        if (!hyperbridgingEnabled[_chainId]) {
+            chainBalance[_chainId] += _amount;
+        }
+    }
+
+    function bridgehubDeposit(
+        uint256 _chainId,
+        address _prevMsgSender,
+        bytes calldata _data
+    ) external payable override onlyBridgehub returns (L2TransactionRequestTwoBridgesInner memory request) {
+        (address _l1Token, uint256 _amount, address _l2Receiver) = abi.decode(_data, (address, uint256, address));
```

Comment:
`_l1Token` isn't used, should we check that it is exactly `l1WethAddress`?

Reply:
added